Subaxis2 for issue #409

ngs333 opened the pull request:
**Description**
Fixes errors in writing regional diagnostics using mask tables by registering global attribute "NumFilesInSet" and per-axis attribute "domain_decomposition" for regional X and Y axis.
NumFilesInSet is always set equal to 0.

Fixes # 409

**How Has This Been Tested?**
Ocean model output  for regional diagnostic with mask tables were performed on Xanadu and 2020.02 code base (plus this fix) and compared.

**Checklist:**
- [ X] My code follows the style guidelines of this project
- [ X] I have performed a self-review of my own code
- [ X] I have commented my code, particularly in hard-to-understand areas
- [ X] I have made corresponding changes to the documentation
- [ X] My changes generate no new warnings
- [ X] Any dependent changes have been merged and published in downstream modules
- [ X] New check tests, if applicable, are included
- [ X] `make distcheck` passes

ngs333 committed:
Fixes regional diagnostic error (issue # 409) by adding global attribute NumFilesInSet and per-axis attribute domain_decomposition for regional X and Y axis.

thomas-robinson commented:
@ngs333 You need to submit this to [NOAA-GFDL/FMS](https://github.com/NOAA-GFDL/FMS/pulls) not my fork.

ngs333 committed:
diag_output.F90 modified to register global attribute "NumFilesInSet" and
per-axis attribute "domain_decomposition" for regional X and Y axis.
NumFilesInSet is always set equal to 0. Fixes errors in writing regional
diagnostics using mask tables.

ngs333 committed:
Adding code comments in diag_output.F90 concerning the registering of NumFilesInSet
and domain_decomposition attributes.

ngs333 committed:
Function write_axis_meta_data of diag_output.F90 now registers attributes NumFilesInSet
and domain_decomposition for regional subaxis. Presence of NumFileInSet is checked so as
to add it only once. Relevant comments are added. This solves github issue  # 409.

ngs333 committed:
Function write_axis_meta_data of diag_output.F90 now registers attributes NumFilesInSet
and domain_decomposition for regional subaxis. Presence of NumFileInSet is checked so as
to add it only once. Relevant comments are added. This solves github issue  # 409.

ngs333 committed:
File diag_output.F90 is modified to register global attribute "NumFilesInSet"
oce per regional diag file - in function diag_output_init. Function write_axis_meta_data is modified to register the attribute "domain_decomposition" for regional X and Y axis. NumFilesInSet is always set equal to 0. Appropriate comments are added.

colingladueNOAA committed:
Merge branch 'master' into subaxis2
